common/persistence/clusterMetadataStore.go

```diff
@@ -206,9 +206,16 @@ func (m *clusterMetadataManagerImpl) convertInternalGetClusterMetadataResponse(
 
 // immutableFieldsChanged returns true if any of immutable fields changed.
 func immutableFieldsChanged(old persistencespb.ClusterMetadata, cur persistencespb.ClusterMetadata) bool {
-	return (old.ClusterName != "" && old.ClusterName != cur.ClusterName) ||
+	if (old.ClusterName != "" && old.ClusterName != cur.ClusterName) ||
```

Comment:
what about clusterName? Should you allow clusterName change if globNamespace is not enabled yet on current cluster.

Reply:
Cluster name is also stored in namespace table so probably not a good idea to change it?

Reply:
Yes. This is the tricky one. We need some offline discussion.

Reply:
👍